src/qz/ws/PrintSocketServer.java

```diff
@@ -154,15 +165,23 @@ public void configure(WebSocketServletFactory factory) {
                         factory.getPolicy().setMaxTextMessageSize(MAX_MESSAGE_SIZE);
                     }
                 };
-                server.setHandler(wsHandler);
+                StatisticsHandler statsHandler = new StatisticsHandler();
+                statsHandler.setHandler(wsHandler);
+                server.setHandler(statsHandler);
+                server.setStopTimeout(1000);
                 server.setStopAtShutdown(true);
                 server.start();
 
                 running.set(true);
                 trayManager.setServer(server, running, securePortIndex, insecurePortIndex);
                 log.info("Server started on port(s) " + TrayManager.getPorts(server));
 
+
                 server.join();
+                while (!server.isStopped()) {
```

Comment:
This added loop doesn't seem necessary, as from what I can gleam from `server.join` it takes care of exactly what this loop has been tasked with doing.

Reply:
It is necessary. I know, it doesn't quite make sense on face value, but without it you get:

```java
java.lang.IllegalStateException: !STOPPED
	at org.eclipse.jetty.server.handler.HandlerWrapper.destroy(HandlerWrapper.java:131)
	at qz.ws.PrintSocketServer.runServer(Unknown Source)
	at qz.ws.PrintSocketServer.main(Unknown Source)
```

There's also a `server.isStopping()` method. I believe `server.join()` will return the moment you call `server.stop()`, but internally the state is `STOPPING`, not `STOPPED`. So we have to wait until it is fully stopped before calling `server.destroy()`.